exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/unorderedreceiver/UnorderedReceiverBatch.java

```diff
@@ -153,8 +153,10 @@ private RawFragmentBatch getNextBatch() throws IOException {
   public IterOutcome next() {
     batchLoader.resetRecordCount();
     stats.startProcessing();
+
+    RawFragmentBatch batch = null;
     try{
-      RawFragmentBatch batch;
+
```

Comment:
Consider moving try block to a separate function.

Reply:
create function `getNextNotEmptyBatch()` that calls `getNextBatch()` and either returns not empty batch or `null`.

Reply:
will do. Thanks for the suggestion.